### ISSUE TITLE
powervm_lpar_instance state=facts doesn't show NPIV WWN's

### ISSUE THREAD

**Issue by BladesOfSteele** [IBM/ansible-power-hmc#129]
When I run:
      - name: Capture LPAR data
        ibm.power_hmc.powervm_lpar_instance:
            hmc_host: '{{ inventory_hostname }}'
            hmc_auth: "{{ curr_hmc_auth }}"
            system_name: '{{ system_name }}'
            vm_name: "{{ lpar }}"
            state: facts
            advanced_info: true
        register: lpar_data

lpar_data does not contain any NPIV WWN's:
            "VirtualFiberChannelAdapters": [],

Running _lssyscfg -r prof -m <system> --filter lpar_ids=<id> -F virtual_fc_adapters_ returns WWN's.

Was hoping to use the facts for zoning and masking.

**Comment by naveenkandakur** [comment]
powervm_lpar_instance module facts state shows only Lpar's info, looking at the command you mentioned above which shows the WWN's is to show the profile info, Could you confirm that you have activated the same profile in the lpar?

**Comment by BladesOfSteele** [comment]
No, the LPAR wasn't activated. Here's the flow I'm using, not sure how else I would do this.

1. Use powervm_lpar_instance to create an LPAR, specifying numerous NPIV adapters.
2. Capture the LPAR data, including WWPN's to feed into SAN switch zoning and storage host masking playbooks. 
3. Allocate storage
4. Then launch the install playbook, which would activate the LPAR.

This is all doable, if I use lssycfg or lsnportlogin commands to capture the WWN's of the created LPAR's. I would think it would be nice to have it doable via the powervm_lpar_instance module.  Seems appropriate.

**Comment by 0rphu** [comment]
Strange, I use the flow you described and it works for me. One difference - after I create the LPAR using the module, I modify processor compatibility mode in lpar profile using hmc_command:

`chsyscfg -r prof -m {{ inventory_hostname }} -i \"lpar_name={{ VM_Name }},name={{ prof_name }},lpar_proc_compat_mode={{ compat }}\" --force`

But I'm pretty sure it worked without this as well. 

Did you check after you create the lpar using lpar_instance if all vfchosts on vios are mapped to physical ports?

**Comment by BladesOfSteele** [comment]
strange is right, when I test it in my lab now, works fine. But in the customer environment, no matter what I tried, didn't work. Might need to capture some version details and see if there's any explanation. 

As for the mapping question, they are for sure mapped, since I can zone and see storage through them.

**Comment by naveenkandakur** [comment]
@BladesOfSteele Please try the scenario with latest version v1.14.0

**Comment by opokam** [comment]
Have you tried applying the profile before gathering facts?

Ie: chsyscfg -r lpar -m <ManagedSystemName> -o apply -n <ProfileName>

**Comment by BladesOfSteele** [comment]
Unfortunately, given this is now a customer's production environment, I cannot test. But will, given the next opportunity on a similar configuration, test the process again.

@opokam the idea is that the module would work without additional manual steps. There's plenty of ways around the issue.
